pages/learn/deploy/deployment.md

```diff
@@ -1,52 +1,250 @@
 
-Alternatively, if your project includes a package.json (i.e. a Node.js project), then that will be used instead to create an implicit Dockerfile, which simulates the build process a node.js/npm project expects. In this way, we are able to transparently run node.js projects on {{ $names.company.lower }}, while taking advantage of some of Docker’s caching features.
+It should be noted that `balena push` is independent of git, so you are free to use any version control system you wish. This also means that it is possible to use [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules) in your project when deploying with `balena push`.
 
-A Dockerfile will always give you more power to fine-tune the build process, but it’s also good to start fast without one, since you can shift to a Dockerfile whenever you like. We will soon add more languages to the “first-class” support we give to Node.js. Until then, any other language can be used by adding a Dockerfile. It’s important to note that the environment within which your code gets built will match the devices you use for your app. So if you’re for example pushing to an app containing BeagleBone Black devices, we’ll build your code in an ARMv7 environment, while for the Raspberry Pi B+ it would be ARMv6.
+### Additional Options
+#### `--source, -s <source>`                    
 
-If you push a project with only a `Dockerfile`, `Dockerfile.template`, or `package.json`, a single container image will be built and sent to your device. The single container will show up on the device dashboard as a service with the name `main`. 
+The `--source` flag allows you do define a path to your source code folder. This flag directs `push` to send that directory to be built on either the [cloud builder](#the-balenacloud-build-server) of [local device](/learn/develop/local-mode/). You should ensure your folder follows the [standard balena project structure](#project-structure).
 
-For [multicontainer][multicontainer] applications (Microservices and Starter [application types][app-types]), a `docker-compose.yml` file at the root of the project directory will kick off multiple simultaneous image builds, each with their own [build logs][logs].
+#### `--emulated, -e` 
+
+The `--emulated` flag will force the [balenaCloud builder](#the-balenacloud-build-server) to run an [qemu](https://www.qemu.org/) emulated build. What this means is that your build will be executed on a `x86_64` CPU that emulates the target architecture of your application, rather than running on the native architecure of your device. You can see the build is emulated in the first few lines of the builder output as below:
```

Comment:
"What this means is that your build will be executed" -> "This means that your build will be executed."

"You can see the build is emulated" -> "You can see if the build is emulated"